src/demo/previewer.js

```diff
@@ -15,139 +16,147 @@ if (process.env.NODE_ENV === 'development') {
-    port1.postMessage({
-      type: 'RENDER',
-      options: this.state,
-    })
+      if (message.data.type === 'SDK_COMPLETE') {
+        setSdkFlowCompleted(true)
 
-  globalUpdateOptionsCall = ({ onComplete, ...sdkOptions }) => {
-    if (onComplete) this.globalOnCompleteFunc = onComplete
-    this.updateSdkOptions(sdkOptions)
-  }
+        if (globalOnCompleteFunc) {
+          globalOnCompleteFunc(message.data.data)
+        }
 
-  updateViewOptions = (options) =>
-    this.setState((prevState) => ({
-      viewOptions: {
-        ...prevState.viewOptions,
-        ...options,
-      },
-    }))
+        console.log('Complete with data!', message.data.data)
+      }
+    }
 
-  updateSdkOptions = (options) =>
-    this.setState((prevState) => ({
-      sdkOptions: {
-        ...prevState.sdkOptions,
-        ...options,
-      },
-    }))
+    const onIFrameLoad = () => {
+      // Transfer port2 to the iframe
+      iframe.current.contentWindow.postMessage('init', '*', [channel.port2])
+    }
 
-  onIFrameLoad = () => {
-    // Transfer port2 to the iframe
-    this.iframe.contentWindow.postMessage('init', '*', [channel.port2])
-  }
-
-  render() {
-    return (
-      <div className="previewer">
-        <div
-          className={`iframe-wrapper${
-            this.state.viewOptions.darkBackground ? ' dark' : ''
-          }`}
-        >
-          <iframe
-            src={`/index.html${window.location.search}`}
-            ref={(iframe) => (this.iframe = iframe)}
-            style={{
-              width: this.state.viewOptions.iframeWidth,
-              height: this.state.viewOptions.iframeHeight,
-            }}
-          />
-        </div>
-        <div className="sidebar">
-          <a href={`/`}>(view vanilla SDK demo page)</a>
+    window.updateOptions = ({ onComplete, ...sdkOptions }) => {
+      if (onComplete) {
+        globalOnCompleteFunc = onComplete
+      }
+      updateSdkOptions(sdkOptions)
+    }
+    window.addEventListener('message', onMessage)
+    port1.onmessage = onMessage
 
-          <SdkOptions
-            sdkOptions={this.state.sdkOptions}
-            updateSdkOptions={this.updateSdkOptions}
-          />
+    const iframeRef = iframe.current
+    if (iframeRef) {
+      iframeRef.addEventListener('load', onIFrameLoad)
+    }
+
+    return () => {
+      delete window.updateOptions
+      delete globalOnCompleteFunc.current
+      window.removeEventListener('message', onMessage)
+
+      if (iframeRef) {
+        iframeRef.removeEventListener('load', onIFrameLoad)
+      }
+    }
+  }, []) // eslint-disable-line react-hooks/exhaustive-deps
 
-          <ViewOptions
-            viewOptions={this.state.viewOptions}
-            updateViewOptions={this.updateViewOptions}
+  /**
+   * Re-render the demo app only if `sdkOptions` or `viewOptions` changed
+   */
+  useEffect(() => {
+    port1.postMessage({
+      type: 'RENDER',
+      options: {
+        checkData,
+        sdkFlowCompleted,
```

Comment:
Where is `checkData`, `sdkFlowCompleted` defined/passed down from?

Reply:
They're hooks defined here: https://github.com/onfido/onfido-sdk-ui/blob/35bd4bd14ae9aeff5e4a59339871166ff90ce59b/src/demo/previewer.js#L27